rlp.nim

```diff
@@ -184,7 +184,7 @@ proc toInt*(self: Rlp, IntType: typedesc): IntType =
     raise newException(BadCastError, "")
 
   for i in payloadStart ..< (payloadStart + payloadSize):
-    result = (result shl 8) or int(self.bytes[self.position + i])
+    result = cast[IntType](result shl 8) or cast[IntType](self.bytes[self.position + i])
```

Comment:
Shouldn't this be a regular conversion (not a cast). `toInt` is supposed to also work with user-defined `BigInt` types and these casts won't be appropriate for them.

Reply:
Yes, I remember I tried a lot of stuff for that line, obviously I somehow missed toInt